src/codeu/chat/server/JSON.java

```diff
@@ -250,11 +246,46 @@ public Model readFromFile (String file) throws IOException
                     Time time = null;
                     String name = "";
                     Uuid id = null;
+                    long updateTime = -1;
+                    HashMap<Uuid, Integer> conversationInterests = new HashMap<>();
+                    ArrayList<Uuid> userInterests = new ArrayList<Uuid>();
+                    boolean conversationInterestsRun = false;
+                    boolean userInterestsRun = false;
                     while (jp.nextToken() != JsonToken.END_ARRAY) {
                         if (jp.getText().equals("name")) {
                             jp.nextToken();
                             name = jp.getText();
                         }
+                        if (jp.getText().equals("conversationInterests")) {
+                            conversationInterestsRun = true;
+                            jp.nextToken();
+                            while(jp.nextToken()!=JsonToken.END_OBJECT)
+                            {
+                                Uuid convoId = null;
+                                try {
+                                    convoId = Uuid.parse(jp.getText());
+                                } catch (IOException e) {
+                                    System.out.println("Invalid Uuid");
```

Comment:
Is this an expected case? If so make sure you have tests for this. If it should never happen, then maybe we want to do something more severe if it does happen? Similar comments for other catch blocks.